### PR TITLE
[10.0] add sale promotion rule based on product assortment

### PR DESCRIPTION
**Pull request by lmignon**
- [X] includes #729 

This PR based on #729 add a new add sale_promotion_rule_assortment. Thanks to this new addon your are able to reduce the appliance of a promotion rule to a set of products.

### CODE REVIEW

File: sale_promotion_rule/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Copyright 2017 Akretion (http://www.akretion.com)
+# Benoît GUILLOT <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Sale Promotion Rule",
+    "summary": "Module to manage promotion rule on sale order",
+    "version": "10.0.1.0.0",
+    "development_status": "Beta",
+    "category": "Sale",
+    "website": "https://akretion.com",
```
Comment (nikul-serpentcs) [inline sale_promotion_rule/__manifest__.py]: @lmignon IMO Replace URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

File: sale_promotion_rule/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    'author': 'Anybox, ACSONE SA/NV,Odoo Community Association (OCA)',
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "external_dependencies": {
```
Comment (nikul-serpentcs) [inline sale_promotion_rule/__manifest__.py]: If not dependencies then remove it

File: sale_promotion_rule/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        "python": [],
+        "bin": [],
+    },
+    "depends": [
+        "sale",
+    ],
+    "data": [
```
Comment (nikul-serpentcs) [inline sale_promotion_rule/__manifest__.py]: sequence as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L29)

File: sale_promotion_rule/tests/test_promotion.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp.tests.common import TransactionCase
```
Comment (nikul-serpentcs) [inline sale_promotion_rule/tests/test_promotion.py]: odoo

File: sale_promotion_rule/tests/test_promotion.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+from openerp.exceptions import Warning as UserError
```
Comment (nikul-serpentcs) [inline sale_promotion_rule/tests/test_promotion.py]: same here

File: sale_promotion_rule/views/sale_promotion_rule.xml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                <field name="minimal_amount"/>
+            </tree>
+        </field>
+    </record>
+
+    <record id="view_sale_promotion_rule_form" model="ir.ui.view">
+        <field name="model">sale.promotion.rule</field>
+        <field name="arch" type="xml">
+            <form string="Promotion Rule">
+                <header>
```
Comment (nikul-serpentcs) [inline sale_promotion_rule/views/sale_promotion_rule.xml]: remove

File: sale_promotion_rule_assortment/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+{
+    'name': 'Sale Promotion Rule Assortment',
+    'summary': """
+        This addon allows to reduce the appliance of a promotion rule to a
+        set of products""",
+    'version': '10.0.1.0.0',
+    "development_status": "Beta",
+    'license': 'AGPL-3',
+    'author': 'ACSONE SA/NV,Odoo Community Association (OCA)',
+    'website': 'https://acsone.eu',
```
Comment (nikul-serpentcs) [inline sale_promotion_rule_assortment/__manifest__.py]: URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: sale_promotion_rule/views/sale_promotion_rule.xml (hunk 6 above)
Comment (lmignon) [reply sale_promotion_rule/views/sale_promotion_rule.xml]: @nikul-serpentcs IMO it's a good practive to declare the header element into the base view to allow easy extension of this part of the view.

File: sale_promotion_rule/__manifest__.py (hunk 1 above)
Comment (lmignon) [reply sale_promotion_rule/__manifest__.py]: @nikul-serpentcs fixed on the base PR #729 https://github.com/OCA/sale-workflow/pull/729/commits/8af44bdbfdf28e800bf6afe8d2791083024d5a03

File: sale_promotion_rule/__manifest__.py (hunk 3 above)
Comment (lmignon) [reply sale_promotion_rule/__manifest__.py]: sequence ?????? What's the problem here? The order of data files is not enforced by the template. The most important IMO is that data files are grouped by type.  I don't plan o change this part of the manifest.

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by rousseldenis** [comment]
@nikul-serpentcs Could you update your review ?

@Cedric-Pigeon @acsonefho 

Could we move forward on this ?

**Comment by lmignon** [comment]
This one works in production for a long time without problem.... Can we move forward? 
ping @sebastienbeau

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1:

**Comment by lmignon** [comment]
@Cedric-Pigeon Can you adapt the tests for your last commit...

**Comment by rafaelbn** [comment]
Please @Cedric-Pigeon could you attend last comment from @lmignon and review? thanks!

**Comment by Cedric-Pigeon** [comment]
@rafaelbn @lmignon Done, thanks.

**Comment by rafaelbn** [comment]
Thank you! Any one knows why bots are failing? :red_circle:  :cry:

**Comment by Cedric-Pigeon** [comment]
@rafaelbn Now everything is green. Can you launch the merge please?

**Comment by rousseldenis** [comment]
@rafaelbn

**Pull request merged**